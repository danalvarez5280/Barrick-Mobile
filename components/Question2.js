import React, { Component } from 'react';
import {
  AppRegistry,
  Button,
  Image,
  Keyboard,
  Slider,
  StatusBar,
  StyleSheet,
  Text,
  TextInput,
  TouchableHighlight,
  View,
} from 'react-native';

class Question2 extends Component {
  constructor() {
    super();
    this.state = {
      riskLevel: 1,
      specificRisk: ''
    }
  }

  handleSubmit() {
    this.props.submitLogin(this.state)
    this.setState({
      riskLevel: 1,
      specificRisk: ''
    })
  };

  noInfo() {
    Alert.alert('You need to input a SPECIFIC TASK')
  };

  render() {
    const submitButton =
      this.state.specificRisk.length ?
        <Button onPress={ this.handleSubmit } title='Submit'/> :
        <Button onPress={ this.noInfo } title='Submit'/>;
    return(
      <View style={styles.question}>
        <Text>Question 2</Text>
        <Slider
        value={this.state.riskLevel}
        onValueChange={(riskLevel) => this.setState({riskLevel})}/>
      </View>
    )
  }
};
const styles = StyleSheet.create({
  question: {
    display: 'flex',
    flexDirection: 'column',
    marginTop: 70,
  },
});

export default Question2;
