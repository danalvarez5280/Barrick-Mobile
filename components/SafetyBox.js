import React, { Component } from 'react';
import {
  AppRegistry,
  Button,
  Image,
  Keyboard,
  Slider,
  StatusBar,
  StyleSheet,
  Text,
  TextInput,
  TouchableHighlight,
  View,
} from 'react-native';

import { KeyboardAwareScrollView } from 'react-native-keyboard-aware-scroll-view';

import safetyChecked from './assets/checked.png';
import notChecked from './assets/not-checked.png';

//colors
// gold #C4900F
// blue #122732
//off white #CED5E0
// off blue #324A5E
class SafetyBox extends Component {
  constructor(props) {
    super();
    this.state = {
      checked: props.checked,
      list: props.list,
      check: props.safeCheck,
    }
    this.changeColor = this.changeColor.bind(this);
  };

  changeColor() {
    console.log('hi dan');
    this.state.checked ?
    this.setState({
      checked: false,
    }) :
    this.setState({
      checked: true,
    })
  };


  render() {
    let boxChecked = this.state.checked ? safetyChecked : notChecked;
    let boxStyle = this.state.checked ? styles.gold : styles.blue;
    return(
      <TouchableHighlight onPress={ () => this.changeColor() }>
        <View style={ boxStyle }>
          <Image source={ boxChecked } style={ styles.checkBox } />
          <Text style={ styles.white }>{ this.state.check }</Text>
        </View>
      </TouchableHighlight>
    )
  }
};

const styles = StyleSheet.create({
  white: {
    color: 'white',
    fontSize: 12,
    fontWeight: 'bold',
    marginLeft: 20,
  },
  blue: {
    alignItems: 'center',
    backgroundColor: '#122732',
    borderRadius: 15,
    display: 'flex',
    flexDirection: 'row',
    height: 40,
    marginTop: 10,
    padding: 5,
    width: 220,
  },
  gold: {
    alignItems: 'center',
    backgroundColor: '#C4900F',
    borderRadius: 15,
    display: 'flex',
    flexDirection: 'row',
    height: 40,
    marginTop: 10,
    padding: 5,
    width: 220,
  },
  checkBox: {
    height: 24,
    width: 24,
  }
});

export default SafetyBox;
