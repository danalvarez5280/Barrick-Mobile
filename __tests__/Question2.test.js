import 'react-native';
import React from 'react';
import { shallow, mount } from 'enzyme';
import Question2 from '../components/Question2';
import SafetyBox from '../components/SafetyBox';
import { Button, Image, Slider, Text, TextInput, TouchableHighlight } from 'react-native';

import renderer from 'react-test-renderer';

const q2Submit = jest.fn();
const howRisky = jest.fn();
const risksChecked = jest.fn();
const logOut = jest.fn();
const goBack = jest.fn();
const mockRiskList = ['grab hard hat', 'be safe'];
const mockState = {
  riskLevel: 1,
  specificRisk: 'working near gas line',
  risksList: ['grab hard hat', 'be safe']
};

const setup = () => {
  return shallow(<Question2
    submitLogin={ q2Submit }
    goBack={ goBack }
    risksChecked={ risksChecked }
    risksList={ mockRiskList }
    info={ mockState }
    logOut={ logOut } />)
};

it('renders correctly', () => {

  const tree = renderer.create(
    <Question2
    submitLogin={ q2Submit }
    goBack={ goBack }
    risksChecked={ risksChecked }
    risksList={ mockRiskList }
    info={ mockState }
    logOut={ logOut } />
  ).toJSON();
  expect(tree).toMatchSnapshot();
});

it('should have one icon rendered', () => {
  let question2 = setup();

  expect(question2.find(Image).length).toEqual(1);
});

it('should have 4 texts displayed', () => {
  let question2 = setup();

  expect(question2.find(Text).length).toEqual(4);
});

it('should have a text input area', () => {
  let question2 = setup();

  expect(question2.find(TextInput).length).toEqual(1);
});

it('should have 2 safety check boxes', () => {
  let question2 = setup();

  expect(question2.find(SafetyBox).length).toEqual(2);
});

it('should be able to log out', () => {
  let question2 = setup();

  expect(question2.find(Button).length).toEqual(3);
  question2.find(Button).last().simulate('press');
  expect(logOut).toHaveBeenCalled();
});

it('should be able to submit the information', () => {
  let question2 = setup();

  expect(question2.find(Button).length).toEqual(3);
  question2.find(Button).first().simulate('press');
  expect(q2Submit).toHaveBeenCalled();
  expect(risksChecked).toHaveBeenCalled();
});

it('should be able to submit the information', () => {
  let question2 = setup();

  expect(question2.find(Button).length).toEqual(3);
  question2.find(Button).at(1).simulate('press');
  expect(goBack).toHaveBeenCalled();
});

it.skip('should have a slider bar to set the risk', () => {
  let question2 = setup();

  expect(question2.find(Slider).length).toEqual(1);
  question2.find(Slider).simulate('ValueChange');
  expect(howRisky).toHaveBeenCalled();
});

it.skip('should set an icon to the page based on the risk level', () => {
  let question2 = setup();

  question2.instance().howRisky();
});
