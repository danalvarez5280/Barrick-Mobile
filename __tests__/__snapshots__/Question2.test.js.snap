// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<RCTScrollView
  automaticallyAdjustContentInsets={false}
  contentContainerStyle={
    Object {
      "display": "flex",
      "flexDirection": "column",
      "height": 670,
      "justifyContent": "space-between",
      "marginTop": 0,
      "padding": 0,
      "width": "100%",
    }
  }
  contentInset={
    Object {
      "bottom": 0,
    }
  }
  enableAutoAutomaticScroll={true}
  enableResetScrollToCoords={true}
  extraHeight={75}
  extraScrollHeight={0}
  keyboardDismissMode="interactive"
  keyboardOpeningTime={250}
  onScroll={[Function]}
  scrollEnabled={true}
  scrollEventThrottle={0}
  showsVerticalScrollIndicator={true}
>
  <View>
    <View
      style={
        Object {
          "alignItems": "center",
          "display": "flex",
          "flexDirection": "column",
          "justifyContent": "center",
          "marginTop": 0,
        }
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "backgroundColor": "#C4900F",
            "color": "#122732",
            "fontWeight": "bold",
            "marginTop": 10,
            "padding": 5,
            "textAlign": "center",
            "width": 250,
          }
        }
      >
        Select the Risk Level
      </Text>
      <Image
        source={
          Object {
            "testUri": "../../../components/assets/shield.png",
          }
        }
        style={
          Object {
            "height": 80,
            "margin": 15,
            "width": 80,
          }
        }
      />
      <RCTSlider
        disabled={false}
        enabled={true}
        maximumTrackTintColor="green"
        maximumValue={7}
        minimumTrackTintColor="red"
        minimumValue={1}
        onChange={[Function]}
        onResponderTerminationRequest={[Function]}
        onSlidingComplete={[Function]}
        onStartShouldSetResponder={[Function]}
        onValueChange={[Function]}
        step={1}
        style={
          Array [
            Object {
              "height": 40,
            },
            Object {
              "width": 200,
            },
          ]
        }
        value={1}
      />
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
      >
        1
      </Text>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "backgroundColor": "#C4900F",
            "color": "#122732",
            "fontWeight": "bold",
            "marginTop": 10,
            "padding": 5,
            "textAlign": "center",
            "width": 250,
          }
        }
      >
        Specific Risks
      </Text>
      <TextInput
        allowFontScaling={true}
        multiline={true}
        onChangeText={[Function]}
        placeholder="Specific Risk"
        style={
          Object {
            "backgroundColor": "#CED5E0",
            "color": "#122732",
            "fontSize": 20,
            "height": 50,
            "marginBottom": 10,
            "padding": 5,
            "width": 250,
          }
        }
        value="working near gas line"
      />
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "backgroundColor": "#C4900F",
            "color": "#122732",
            "fontWeight": "bold",
            "marginTop": 10,
            "padding": 5,
            "textAlign": "center",
            "width": 250,
          }
        }
      >
        Other Risks
      </Text>
      <View
        accessibilityComponentType={undefined}
        accessibilityLabel={undefined}
        accessibilityTraits={undefined}
        accessible={true}
        hasTVPreferredFocus={undefined}
        hitSlop={undefined}
        isTVSelectable={true}
        nativeID={undefined}
        onLayout={undefined}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Array [
            Object {
              "backgroundColor": "transparent",
            },
            undefined,
          ]
        }
        testID={undefined}
        tvParallaxProperties={undefined}
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "#122732",
              "borderRadius": 15,
              "display": "flex",
              "flexDirection": "row",
              "height": 40,
              "marginTop": 10,
              "padding": 5,
              "width": 225,
            }
          }
        >
          <Image
            source={
              Object {
                "testUri": "../../../components/assets/not-checked.png",
              }
            }
            style={
              Object {
                "height": 24,
                "width": 24,
              }
            }
          />
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Object {
                "color": "white",
                "fontSize": 12,
                "fontWeight": "bold",
                "marginLeft": 20,
              }
            }
          >
            grab hard hat
          </Text>
        </View>
      </View>
      <View
        accessibilityComponentType={undefined}
        accessibilityLabel={undefined}
        accessibilityTraits={undefined}
        accessible={true}
        hasTVPreferredFocus={undefined}
        hitSlop={undefined}
        isTVSelectable={true}
        nativeID={undefined}
        onLayout={undefined}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Array [
            Object {
              "backgroundColor": "transparent",
            },
            undefined,
          ]
        }
        testID={undefined}
        tvParallaxProperties={undefined}
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "#122732",
              "borderRadius": 15,
              "display": "flex",
              "flexDirection": "row",
              "height": 40,
              "marginTop": 10,
              "padding": 5,
              "width": 225,
            }
          }
        >
          <Image
            source={
              Object {
                "testUri": "../../../components/assets/not-checked.png",
              }
            }
            style={
              Object {
                "height": 24,
                "width": 24,
              }
            }
          />
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Object {
                "color": "white",
                "fontSize": 12,
                "fontWeight": "bold",
                "marginLeft": 20,
              }
            }
          >
            be safe
          </Text>
        </View>
      </View>
      <View
        accessibilityComponentType="button"
        accessibilityLabel={undefined}
        accessibilityTraits={
          Array [
            "button",
          ]
        }
        accessible={true}
        collapsable={undefined}
        hitSlop={undefined}
        isTVSelectable={true}
        nativeID={undefined}
        onLayout={undefined}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
        testID={undefined}
        tvParallaxProperties={undefined}
      >
        <View
          style={
            Array [
              Object {},
            ]
          }
        >
          <Text
            accessible={true}
            allowFontScaling={true}
            disabled={undefined}
            ellipsizeMode="tail"
            style={
              Array [
                Object {
                  "color": "#007AFF",
                  "fontSize": 18,
                  "padding": 8,
                  "textAlign": "center",
                },
              ]
            }
          >
            SUBMIT
          </Text>
        </View>
      </View>
      <View
        accessibilityComponentType="button"
        accessibilityLabel={undefined}
        accessibilityTraits={
          Array [
            "button",
          ]
        }
        accessible={true}
        collapsable={undefined}
        hitSlop={undefined}
        isTVSelectable={true}
        nativeID={undefined}
        onLayout={undefined}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
        testID={undefined}
        tvParallaxProperties={undefined}
      >
        <View
          style={
            Array [
              Object {},
            ]
          }
        >
          <Text
            accessible={true}
            allowFontScaling={true}
            disabled={undefined}
            ellipsizeMode="tail"
            style={
              Array [
                Object {
                  "color": "#007AFF",
                  "fontSize": 18,
                  "padding": 8,
                  "textAlign": "center",
                },
              ]
            }
          >
            BACK
          </Text>
        </View>
      </View>
      <View
        accessibilityComponentType="button"
        accessibilityLabel={undefined}
        accessibilityTraits={
          Array [
            "button",
          ]
        }
        accessible={true}
        collapsable={undefined}
        hitSlop={undefined}
        isTVSelectable={true}
        nativeID={undefined}
        onLayout={undefined}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
        testID={undefined}
        tvParallaxProperties={undefined}
      >
        <View
          style={
            Array [
              Object {},
            ]
          }
        >
          <Text
            accessible={true}
            allowFontScaling={true}
            disabled={undefined}
            ellipsizeMode="tail"
            style={
              Array [
                Object {
                  "color": "#007AFF",
                  "fontSize": 18,
                  "padding": 8,
                  "textAlign": "center",
                },
              ]
            }
          >
            LOG OUT
          </Text>
        </View>
      </View>
    </View>
  </View>
</RCTScrollView>
`;
